### DISCUSSION TITLE
Shows React component information from functional render function

### DISCUSSION THREAD

**Discussion by bgirard** [RecordReplay/devtools#6092] [Ideas]
**Description:**
When you're within a React render function:
<img width="595" alt="Screen Shot 2022-03-30 at 8 20 46 PM" src="https://user-images.githubusercontent.com/793565/160969796-bb13c0e4-abe8-4b6e-951d-9e6248396c57.png">
You don't know if the component is on it's first mount (or remounting), or if it's a re-render. The lifecycle implication can be important when root causing a bug:
![image](https://user-images.githubusercontent.com/793565/160969924-ec9f0995-c87d-4d9b-b9ba-80368897b338.png)

Usually that I want to know is did this component accidentally remount when it shouldn't have so I can look at the parent to find out why, or is it another problem.

I can do this manually but it's very tedious. I go up the stack to React calling the render function. Locate the FiberNode and check for `alternate`.
First mount:
<img width="364" alt="Screen Shot 2022-03-30 at 8 24 12 PM" src="https://user-images.githubusercontent.com/793565/160970155-22effd9b-b189-4809-a891-a7c4539dfc33.png">

Re-render:
<img width="373" alt="Screen Shot 2022-03-30 at 8 23 39 PM" src="https://user-images.githubusercontent.com/793565/160970123-ff60901a-4dda-40e3-9b3e-3cd9d98f1469.png">

It would be nice to have a feature to help with this. Either let me run a small expression and plug it into the UI. This is harder then a typically console because I have to change the stack frame.

Or these feature could be built in. Basically go up the stack, look for an object that has the fiber properties (alternate, lastEffect, etc...) and report relevant one.

**Comment by jasonLaster** [comment]
This ties in really nicely to react component stacks because it helps explain the component lifecycle
